lightning/src/util/wakers.rs

```diff
@@ -172,6 +174,16 @@ impl Future {
 			state.callbacks.push(callback);
 		}
 	}
+
+	// C bindings don't (currently) know how to map `Box<dyn Trait>`, and while it could add the
+	// following wrapper, doing it in the bindings is currently much more work than simply doing it
+	// here.
+	/// Registers a callback to be called upon completion of this future. If the future has already
+	/// completed, the callback will be called immediately.
+	#[cfg(c_bindings)]
+	pub fn register_callback_fn<F: 'static + FutureCallback>(&self, callback: F) {
```

Comment:
Looks good. I can't think of a better name, to be honest. It is a shame to have the discrepancy in naming between bindings/using Rust directly though, but this change is pretty simple and so probably best for now.